src/prefect/server/models/block_documents.py

```diff
@@ -414,6 +430,30 @@ async def read_block_documents(
     return fully_constructed_block_documents
 
 
+@inject_db
+async def count_block_documents(
+    session: AsyncSession,
+    db: PrefectDBInterface,
+    block_document_filter: Optional[schemas.filters.BlockDocumentFilter] = None,
+    block_type_filter: Optional[schemas.filters.BlockDocumentFilter] = None,
+    block_schema_filter: Optional[schemas.filters.BlockDocumentFilter] = None,
```

Comment:
nit: I think these types can be tightened to `Optional[schemas.filters.BlockTypeFilter]` and `Optional[schemas.filters.BlockSchemaFilter]`